snippets/sdk/python/deployments/create_or_update.py

```diff
@@ -0,0 +1,90 @@
+#! /usr/local/env python3
+
+import os
+
+from azure.identity import DefaultAzureCredential
+from azure.mgmt.nginx import NginxManagementClient
+from azure.mgmt.network import NetworkManagementClient
+
+import prerequisites
+
+
+def main():
+    SUBSCRIPTION_ID = os.environ.get("AZURE_SUBSCRIPTION_ID", None)
+    GROUP_NAME = "myResourceGroup"
+    DEPLOYMENT_NAME = "myDeployment"
+
+    # Create clients
+    # For other authentication approaches, please see: https://pypi.org/project/azure-identity/
+    nginx_client = NginxManagementClient(
+        credential=DefaultAzureCredential(), subscription_id=SUBSCRIPTION_ID
+    )
+    # Create prerequisite resources
+    subnet, public_ip, identity = prerequisites.create()
+
+    # - private IP configuration example
+    network_profile_private = {
+        "networkInterfaceConfiguration": {"subnetId": subnet.id},
+        "frontendIPConfiguration": {
+            "privateIPAddresses": [
+                {
+                    "subnetId": subnet.id,
+                    "privateIPAllocationMethod": "Static",
+                    "privateIPAddress": "10.0.1.4",
+                }
+            ]
+        },
+    }
+
+    # - public IP configuration example
+    network_profile_public = {
+        "networkInterfaceConfiguration": {"subnetId": subnet.id},
+        "frontendIPConfiguration": {"publicIPAddresses": [{"id": public_ip.id}]},
+    }
+
+    # Create deployment with a public IP
+    deployment = nginx_client.deployments.begin_create_or_update(
+        GROUP_NAME,
+        DEPLOYMENT_NAME,
+        {
+            "sku": {"name": "preview_Monthly_gmz7xq9ge3py"},
+            "tags": {"myKey": "myValue"},
+            "location": "eastus",
+            "identity": {
+                "type": "UserAssigned",
+                "userAssignedIdentities": {identity.id: {}},
+            },
+            "properties": {
+                "enableDiagnosticSupport": False,
+                "networkProfile": network_profile_public,
+            },
+        },
+    ).result()
+    print("Created a deployment:\n{}".format(deployment))
+
+    # Get deployment
+    deployment = nginx_client.deployments.get(GROUP_NAME, DEPLOYMENT_NAME)
+    print("Get deployment:\n{}".format(deployment))
+
+    # Update deployment with a new tag
+    deployment = nginx_client.deployments.begin_update(
```

Comment:
is there a reason to update tags?

Reply:
Other than to provide an example of how to update a deployment. No.